### ISSUE TITLE
Triggering garbage collector in event handler causes error.

### ISSUE THREAD

**Issue by lleyton** [yue/yue#79]
# OS
MacOS 10.14.2

# Version of yue
0.6.3

# Issue
Trigerring the Lua garbage collector while yue is running an event handler crashes the program and causes an error.

# Steps to reproduce
1. Run the following Lua script using standard lua with the yue library.
2. Click the button labeled "Start GC Cycle"

# Code
```Lua
local gui = require "yue.gui"

local window = gui.Window.create{}

local button = gui.Button.create("Start GC Cycle")

function button.onclick()
    collectgarbage()
end

window:setcontentview(button)
window:setcontentsize{width=400, height=400}
window:center()
window:activate()

gui.MessageLoop.run()
```

# Output/Error
```
objc[7873]: Invalid or prematurely-freed autorelease pool 0x7fc20d82a1d8.
```
# Gif example
![example](https://user-images.githubusercontent.com/26487268/55753465-3c9d8700-59ff-11e9-8135-14f4420119a9.gif)

**Comment by hackwaly** [comment]
@Lleyton-Gray Which terminal theme do you use? So pretty!

**Comment by Lleyton-Gray** [comment]
@hackwaly I use a modified config of [powerlevel9k](https://github.com/bhilburn/powerlevel9k).

**Comment by zcbenz** [comment]
Fixed with a2897ca.
